xwiki-commons-core/xwiki-commons-extension/xwiki-commons-extension-api/src/main/java/org/xwiki/extension/repository/internal/local/LocalExtensionStorage.java

```diff
@@ -306,5 +306,12 @@ public void removeExtension(DefaultLocalExtension extension) throws IOException
         if (extensionFile != null) {
             extensionFile.getFile().delete();
         }
+
+        // Delete extension version folder if empty
+        File extensionVesionFolder = descriptorFile.getParentFile();
+        extensionVesionFolder.delete();
+
+        // Delete extension folder if empty
+        extensionVesionFolder.getParentFile().delete();
     }
```

Comment:
We should also handle the case when `delete()` returns false and at the very least either retry and/or log a warning. The existing code is also lacking this.

Reply:
We should also probably use java.nio.Files, see https://sonarcloud.io/organizations/xwiki/rules?open=java%3AS4042&rule_key=java%3AS4042

Reply:
> We should also handle the case when delete() returns false and at the very least either retry and/or log a warning.

If it returns false, it usually just means it was not empty (which is something expected in many cases).

Reply:
Then I'd say it's a bad logic to call delete() without checking if it's empty.

Reply:
In case that might be useful I give a try to use `java.nio.file` (only for `removeExtension`): https://github.com/xwiki/xwiki-commons/compare/master...amottier:fix-delete-extension-folder-on-removal-nio

If pertinent I can close this PR and open a new one with my `fix-delete-extension-folder-on-removal-nio` branch.

Reply:
I think it's good to apply this PR and then open a second one to improve the nio stuff if you want. Thx, I'll comment on the nio commits.

Reply:
> We should also handle the case when delete() returns false

Note that Sonar and not too happy about that ignored return and fail the build, but https://github.com/xwiki/xwiki-commons/pull/1167 should fix this.